Sources/JWTKit/RSA/RSAKey.swift

```diff
@@ -235,3 +227,24 @@ public struct RSAKey: Sendable {
         }
     }
 }
+
+extension RSAKey: Equatable {
+    public static func == (lhs: RSAKey, rhs: RSAKey) -> Bool {
+        // Compare public keys
+        guard lhs.publicKey.derRepresentation == rhs.publicKey.derRepresentation else {
+            return false
```

Comment:
When a private key is present, comparing the public key is wasteful of CPU cycles. Make the public key compare an assertion (`assert()` specifically, not a `precondition()`, so that it only fires in debug) for that case, and compare it normally when there _isn't_ a private key.